src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -968,32 +1033,29 @@ trait Scanners extends ScannersCommon {
           finishStringPart()
           nextRawChar()
           next.token = USCORE
-        } else if (Character.isUnicodeIdentifierStart(ch)) {
+        } else if (isUnicodeIdentifierStart(ch)) {
           finishStringPart()
-          do {
-            putChar(ch)
-            nextRawChar()
-          } while (ch != SU && Character.isUnicodeIdentifierPart(ch))
-          next.token = IDENTIFIER
-          next.name = newTermName(cbuf.toString)
-          cbuf.clear()
-          val idx = next.name.start - kwOffset
-          if (idx >= 0 && idx < kwArray.length) {
-            next.token = kwArray(idx)
-          }
+          putChar(ch)
+          nextRawChar()
+          getInterpolatedIdentRest()
+        } else if (atSupplementary(ch, isUnicodeIdentifierStart)) {
```

Comment:
Sorry, guy, but this is totally wrong. `s"$id"` requires alpha to start and alnum plus underscore for rest.

Reply:
Actually, why did he say that? it's perfectly fine to test isUnicodeIdentifier. Maybe he was thinking how they rejected accepting arbitrary ids, operator ids and backticks.